### ISSUE TITLE
leechblock not blocking keywords 

### ISSUE THREAD

**Issue by sam12688** [proginosko/LeechBlockNG#390]
In the previous  version , I was able to block keywords on firefox using leechblock, but now after the update it doesn't work

**Comment by proginosko** [comment]
Which OS are you using (Windows, MacOS, etc.)?

Can you provide an example case for me to test?

**Comment by sam12688** [comment]
I am using android device

**Comment by sam12688** [comment]
Like when I am using *(website name).com,  it works perfectly fine

But when using
*
~(keyword)
It doesn't seem to work

**Comment by proginosko** [comment]
I think this related to an issue with Firefox for Android's patchy support for Manifest V3 (which LB adopted in version 1.6).

Try this: go to a site with the keyword that should be blocked. Now open the LeechBlock menu (via the Addons menu). Go back to the site. Is it blocked now? (I'm assuming you have the option "Immediately block pages..." selected.)

**Comment by sam12688** [comment]
> I think this related to an issue with Firefox for Android's patchy support for Manifest V3 (which LB adopted in version 1.6).
> 
> Try this: go to a site with the keyword that should be blocked. Now open the LeechBlock menu (via the Addons menu). Go back to the site. Is it blocked now? (I'm assuming you have the option "Immediately block pages..." selected.)

Yep , the site gets blocked on doing this
But now after closing all the tabs and again opening the site  nothing happens

**Comment by sam12688** [comment]
Is this issue going to stay forever or ever get resolved,  this addon was literally perfect

**Comment by proginosko** [comment]
That depends on how soon the developers of Firefox for Android fix these issues. I've seen reports that LB works better with Firefox Beta, so you might give that a try.

**Comment by gary02** [comment]
firefox 115.7.0esr  in linux has same problem

**Comment by Anton-V-K** [comment]
Keywords blocking still doesn't work in Firefox 115.9.0esr on Windows.

**Comment by proginosko** [comment]
Okay, try this:

Go to **about:debugging**, click **This Firefox**, locate LeechBlock in list of extensions, and click **Inspect**. A "Developer Tools" window should open. Select the **Console** tab, ensure that Errors and Warnings are selected, and click the dustbin icon to clear the console. Now go to a website that should be blocked with a keyword. Do any error messages appear in the console?

**Comment by Anton-V-K** [comment]
@proginosko , thanks for looking into this.
My Firefox 115.9.0esr (32-bit) (Windows 10 21H1 64-bit) shows nothing in Console tab when inspecting LeechBlock NG 1.6.3. BTW Network tab also shows no requests, though I'm not sure whether they should appear there.
LeechBlock NG does block sites by url though (ne requests in Network tab in this case either).

**Comment by proginosko** [comment]
> @proginosko , thanks for looking into this. My Firefox 115.9.0esr (32-bit) (Windows 10 21H1 64-bit) shows nothing in Console tab when inspecting LeechBlock NG 1.6.3. BTW Network tab also shows no requests, though I'm not sure whether they should appear there. LeechBlock NG does block sites by url though (ne requests in Network tab in this case either).

Go to **about:addons** and then to the Permissions page for LeechBlock NG. Is the option "Access your data for all websites" selected? If not, select it and test again.

**Comment by vytevvv** [comment]
@proginosko I went through this thread since as to not create a new issue for the keywords problem on firefox. I am using the Version 125.0.3 (64-bit) have "enabled access your data for all websites" and the console doesn't give anything in return to opening a "should be blocked" page. I have been using LB for the past 3 years on chrome and have switched to firefox couple months ago. Is there a fix yet or a version of firefox I should downgrade to? I can give you more info if needed -  many thanks in advance.

**Comment by proginosko** [comment]
@vytevvv Please export your options to a text file (Options > General > Export Options) and email it to me at leechblockng@proginosko.com so I can take a closer look. Also include details of an example case that I can test.

**Comment by vytevvv** [comment]
Solution: I had "within time periods OR after time limit" without a time limit set. Setting it to "AND" fixed the error as pointed out by @proginosko via email. Best regards!

**Comment by proginosko** [comment]
> Solution: I had "within time periods OR after time limit" without a time limit set. Setting it to "AND" fixed the error as pointed out by @proginosko via email.

Actually, it's the other way around (should be "OR" rather than "AND" if only time periods are specified) -- but glad it was a simple fix!

**Comment by proginosko** [comment]
Update: The bug that affected keyword blocking has been fixed in Firefox Nightly (v128) so hopefully it won't be too long before the fix makes its way to Firefox Beta and then Firefox.

**Comment by Anton-V-K** [comment]
> Is the option "Access your data for all websites" selected? If not, select it and test again.

I can confirm that enabling the option "Access your data for all websites" for LeechBlockNG solves the issue with non-working keywords blocking (in Firefox 115.9.0esr on Windows)
